### PR TITLE
[WIP] Try simple scheduling for xent-regularize

### PR DESCRIPTION
**Pull request by hhadian**

### CODE REVIEW

**Comment by hhadian** [comment]
With `xent-regularize="0.2,0.1"` and linear interpolation I got this:
```
System                  tdnn_7h  tdnn_xentschedule_1a
WER on train_dev(tg)      13.96     13.95
WER on train_dev(fg)      12.86     12.73
WER on eval2000(tg)        16.5      16.5
WER on eval2000(fg)        14.8      14.9
Final train prob     -0.0925208  -0.0908774
Final valid prob      -0.116597   -0.113883
Final train prob (xent)     -1.317    -1.28658
Final valid prob (xent)   -1.40473    -1.37876
```

**Comment by danpovey** [comment]
Looks promising from an objf perspective; boththe regular and xent probs
are better.  Odd that there are no WER gains; could be too small to see.
Try 0.15,0.5.

**Comment by hhadian** [comment]
OK, trying `0.15,0.5`...

**Comment by hhadian** [comment]
`"0.15,0.5" (xentschedule_1b)` vs `"0.2,0.1" (xentschedule_1a)` vs baseline:
```
System                       tdnn_7h    tdnn_xentschedule_1a    tdnn_xentschedule_1b
WER on train_dev(tg)           13.96          13.95          13.98
WER on train_dev(fg)           12.86          12.73          12.93
WER on eval2000(tg)             16.5           16.5           16.9
WER on eval2000(fg)             14.8           14.9           15.2
Final train prob          -0.0925208     -0.0908774     -0.0918364
Final valid prob           -0.116597      -0.113883      -0.115069
Final train prob (xent)             -1.317       -1.28658       -1.26917
Final valid prob (xent)           -1.40473       -1.37876       -1.35618
```

File: egs/wsj/s5/steps/nnet3/chain/train.py [hunk 1]
```diff
@@ -68,8 +68,8 @@ def get_args():
                         without the log-softmax layer for the outputs.  As
                         l2-norm of the log-softmax outputs can dominate the
                         objective function.""")
-    parser.add_argument("--chain.xent-regularize", type=float,
-                        dest='xent_regularize', default=0.0,
+    parser.add_argument("--chain.xent-regularize", type=str,
```
Comment (vijayaditya) [inline egs/wsj/s5/steps/nnet3/chain/train.py]: If you want you could add an action to convert the string to a float array for you. See common_lib.StrToBoolAction and common_lib.NullstrToNoneAction.

Comment (hhadian) [inline egs/wsj/s5/steps/nnet3/chain/train.py]: That would be nicer code-wise. If it works I'll change it. 
Also if it works, we might want to try other complicated schedules where we can set x-values, so we should wait anyway.

**Comment by hhadian** [comment]
`"0.15,0.05"` schedule: 
```
System                       tdnn_7h   tdnn_xentschedule_1c
WER on train_dev(tg)           13.96          14.10
WER on train_dev(fg)           12.86          13.01
WER on eval2000(tg)             16.5           16.8
WER on eval2000(fg)             14.8           15.1
Final train prob          -0.0925208     -0.0914361
Final valid prob           -0.116597      -0.114128
Final train prob (xent)             -1.317       -1.31104
Final valid prob (xent)           -1.40473       -1.39351
```

**Comment by danpovey** [comment]
It looks like this not not very useful :-( closing the pull request but keep the branch around just in case.

**Pull request closed without merging**